sessions/debugging.qmd

```diff
@@ -1,24 +1,192 @@
 # Checking correctness of code: Debugging
 
--   Using basic debugging strategies ('breakpoints', simple `print()` or
-    `stop()` calls, `debug()`, `browser()`)
--   Can take some from the intermediate course (I think that's where it
-    is)
-
 {{< include ../includes/_wip.qmd >}}
 
 ```{r setup}
 #| include: false
 ```
 
-<!-- TODO: Add an introduction here -->
+**Introduction**
+
+Debugging is one of activities that seems really scary and difficult,
+but once you try it and use it, is not nearly as intimidating as it
+seemed. In this session we will focus on using RStudio's interactive
+debugger to debug code.
 
 **Session objectives**:
 
--   Enter the interactive debugger and use it to explore why unexpected
-    behavior is occurring
+-   Understand what debugging is.
+-   Identify how the debugger tool is started in RStudio, what it looks
+    like and apply it to debug functions that cause issues.
+-   Understand `traceback()` error messages to identify which function
+    is causing an error.
+-   Apply `stop()` statements to your functions to protect users from
+    unexpected behavior.
+
+## What is debugging?
```

Comment:
For now I won't review this section, since it will likely change quite a bit as we add the exercises. But great progress!!